### PR TITLE
Visual Studio 2014 support for node v0.10.x series

### PR DESCRIPTION
**Pull request by springmeyer**
Work in progress to get node v0.10.x working against Visual Studio 2014
- [x] figure out what changed with getenv/setenv behavior with 2014 runtimes (https://github.com/mapbox/node-srs/issues/48)
- [ ] try to get gyp 2014 patch accepted upstream - https://code.google.com/p/gyp/issues/detail?id=446&q=2014

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -724,7 +724,7 @@ if options.use_ninja:
 elif options.use_xcode:
   gyp_args += ['-f', 'xcode']
 elif flavor == 'win':
-  gyp_args += ['-f', 'msvs', '-G', 'msvs_version=auto']
+  gyp_args += ['-f', 'msvs', '-G', 'msvs_version=2013']
```
Comment (springmeyer) [inline configure:727]: Tricking gyp by asking for `2013` will not be needed once gyp upstream gets an understanding of 2014: https://code.google.com/p/gyp/issues/detail?id=446&q=2014

File: configure [hunk 2]
```diff
@@ -655,7 +655,7 @@ def configure_winsdk(o):
 
   winsdk_dir = os.environ.get("WindowsSdkDir")
 
-  if winsdk_dir and os.path.isfile(winsdk_dir + '\\bin\\ctrpp.exe'):
+  if winsdk_dir and os.path.isfile(winsdk_dir + '\\bin\\x64\\ctrpp.exe'):
```
Comment (springmeyer) [inline configure:658]: @BergWerkGIS - can you comment on this change ^^. Did you find it optional or mandatory for building against visual studio 2014?

Comment (wilhelmberg) [inline configure:658]: I think it's mandatory, but I haven't investigated very much.

My experience was like this:
- first run without this change -> build stopped with error message. Something like `couldn't find WinSDK or ctrpp`
- inserted `x64` -> build was successful
- second run, where I accidentally had changed `bin\x64` (correct) to `x64\bin` (wrong) -> build was **successfull**

I think some info/data/? gets generated and cached, if `ctrpp.exe` is found. If it is not found afterwards the cached data is used.

Comment (wilhelmberg) [inline configure:658]: @springmeyer 
For 32/64bit builds we should think about how to make the `\\x64\\` part of the path target arch agnostic.
The paths are: `C:\Program Files (x86)\Windows Kits\8.1\bin\[x86|x64]\ctrpp.exe`.
 But I couldn't find variable within `configure` that outputs [x86|x64], as `target_arch` gives  `ia32` for x86 builds.
At least the 32bit builds complete succuessfully, even with `x64\ctrpp.exe`.
Don't know if there will be problem later on.

Comment (springmeyer) [inline configure:658]: Okay - thinking about trying to get this upstream eventually - maybe we could switch to using `C:\Program Files (x86)\Windows Kits\8.1\bin\x64\ctrpp.exe` only if it exists and falling back to `C:\Program Files (x86)\Windows Kits\8.1\bin\ctrpp.exe`? I presume that >= 2014 uses the arch folder but not previous versions?

Comment (wilhelmberg) [inline configure:658]: Y, that difference didn't exist before.

**Comment by bnoordhuis** [comment]
@springmeyer I would recommend sending the patch to the gyp-developers mailing list.  The issue tracker is a graveyard, in my experience.

**Comment by springmeyer** [comment]
@bnoordhuis okay, thank you. My next step is to update it for `2015` (once I've tested) then will email gyp-developers.

**Pull request closed without merging**